frontend/components/S3Settings.vue

```diff
@@ -1,49 +1,114 @@
 <script>
+import axios from 'axios'
+import Messages from './Messages.vue'
+
 export default {
   name: 'S3Settings',
-  props: {
-    buckets: Array,
+  components: {
+    Messages,
+  },
+  data() {
+    return {
+      ready: false,
+      error: null,
+      buckets: [],
+      accessKeyId: null,
+      secretAccessKey: null,
+      bucketName: null,
+    }
+  },
+  methods: {
+    updateBuckets() {
+      axios.get('/s3/api/bucket')
+        .then(res => {
+          this.ready = true
+          this.error = res.data.error
+          this.buckets = res.data.buckets
+        })
+    },
+    addBucket() {
+      const payload = {
+        'access-key-id': this.accessKeyId,
+        'secret-access-key': this.secretAccessKey,
+        'bucket-name': this.bucketName,
+      }
+      axios.post('/s3/api/bucket', payload)
+        .then(res => {
+          this.error = res.data.error
+          if (!this.error) {
+            this.buckets = [...this.buckets, {
+              name: payload['bucket-name'],
+            }]
+          }
+        })
+    },
+    removeBucket(bucket) {
+      axios.get(`/s3/api/bucket/${bucket.name}/remove`)
```

Comment:
I suppose yes we should for deletions. But in this case it's `deactivate` in reality so I need to change it based on your `activate/deactivate` comment.